core/utilities/mixins/_flex-container.scss

```diff
@@ -38,36 +39,41 @@ $flex-container-defaults: (
 /// flex-basis: auto
 /// //  $_order: <integer>  // Note that this should be set at the element level
 /// // $_align-self: flex-start // Note that this should be set at the element level
-@mixin flex-container($settings-map: $flex-container-defaults) {
+@mixin flex-container($settings-map: $default-flex-container) {
 
-  // Pull out the parts for easy use.
-  $parent: map-get($settings-map, 'parent');
-  $children: map-get($settings-map, 'children');
-  $flex: map-get($settings-map, 'flex');
-  $flex-direction: map-get($settings-map, 'flex-direction');
-  $flex-wrap: map-get($settings-map, 'flex-wrap');
-  $justify-content: map-get($settings-map, 'justify-content');
-  $align-items: map-get($settings-map, 'align-items');
-  $align-content: map-get($settings-map, 'align-content');
-  $flex-grow: map-get($settings-map, 'flex-grow');
-  $flex-shrink: map-get($settings-map, 'flex-shrink');
-  $flex-basis: map-get($settings-map, 'flex-basis');
+  // Merge the provided map with the default aggregate from above. This means
+  // that you can specify a container that only has the values that you want to differ
+  // from the default, and everything else is inherited.
+  $_flex-container: map-merge($default-flex-container, $settings-map);
```

Comment:
I'm more of a fan of the simple merge. If someone overrides the default and messes it up that is their problem and scss should spit out errors in order to get their attention to fix it.